### DISCUSSION TITLE
Is the data in the filters secure?

### DISCUSSION THREAD

**Discussion by KristijanZic** [pocketbase/pocketbase#5331] [Q&A]
So I have a PocketBase that has a collection of tokens and I'm logging into the admin and trying to find the token in the collection like this:
`final record = await pb.collection('token_collection').getFirstListItem('token_field="$token"');`
This would return a record if it found one which token_field is the same as our token. Meaning the token is valid if it exits.

I'm worried about putting the token in the filter and I'm wondering how that works. As far as I understood it, the above code puts my token in the URL of the request? Could somebody intercept this request with wireshark and read the token from the URL?

What would be the way to do this securely?
I have to say in advance that I can't have an endpoint by extending PocketBase in Go or JavaScript for business reasons. It has to be a server made in Dart that calls out the PocketBase. Also, I can't hook up the server in dart directly to the DB, it has to be like this. Am I going to be hacked?

**Comment by ganigeorgiev** [comment]
The above expression will be part of the `filter` query parameter (_see the List API details in https://pocketbase.io/docs/api-records/#listsearch-records_) and like any other query or body parameter it will be encrypted when HTTPS is used. It will be available as plain text value in the client history and in the server logs.

If the above is executed in a server context and the `$token` variable comes from untrusted user input then it is vulnerable to string injection attacks and for such cases it is recommended to bind it as parameter (see [Dart SDK - Binding filter parameters](https://github.com/pocketbase/dart-sdk?tab=readme-ov-file#binding-filter-parameters)).

Technically comparing directly security sensitive strings could be vulnerable to timing attacks as an attacker could control one of the strings and its length (_whether in practice this could be exploited I'm not sure since the network response time would vary for a lot of reasons_).

> What would be the way to do this securely?

I don't know what to recommend. There is not enough context to understand the purpose of the `token_collection`, how it affects or relates to other collections and why it is needed in the first place.
